src/verify.c

```diff
@@ -727,9 +727,13 @@ int verify_main(int argc, char **argv)
 
 	grabtime_start(&times, "Load and recurse Manifests");
 
-	/* If upstream URL is passed, user is likely trying to get back to the official stream,
-	 * so ignore any custom mixer related state */
-	if (strncmp(content_url, DEFAULT_CONTENTURL, strlen(DEFAULT_CONTENTURL)) == 0) {
+	/* If upstream URL is passed, user is trying to verify against the official
+	 * stream. The --fix flag is also not supported for mixed systems due to
+	 * special handling of some os-core files being required, which is not
+	 * currently done. For both these cases ignore any custom mixer related
+	 * state */
+	if (cmdline_option_fix ||
```

Comment:
I'm not sure about that. system_on_mix() == true can't we have scenarios where "swupd verify" won't list any problem and "swupd verify --fix" will list and fix files?

Reply:
My 2 cents on that:

I think that if a system is on a mix or not should be completely transparent to swupd. And the same for DEFAULT_CONTENTURL. This should be irrelevant. If a person have is using a mirror of the clearlinux repo, shouldn't we have the exactly same behavior as if the person is downloading strait from the upstream server?

I think we could support https:// and file:// repositories with an order of priority and just don't worry if the repository we are getting info from is a mix or the upstream one. If there's any difference in the files structure in a mix (manifest, different files, sigs, etc) we should be able to guess that from the content and not from any setting in swupd.

Reply:
My comments on the second comment were definitely not blockers for the commit to get in. But I still think that it's possible to have different results if you are running "swupd verify" or "swupd verify --fix". I'll try to check that.

> it is not a mirror of the clearlinux repo. It is the clearlinux update stream plus local content merged in. There are essentially two sources for the update.

I know that a mix is not a mirror. But I think there's no reason not supporting mirrors for Clear Linux, so I don't like comparing the DEFAULT_CONTENTURL to take decisions in swupd. And we don't need that

> but this is not the point of the local mix content. This would mean users would have to host the entire clear update content on their machine.

No. That's why I said we could have several repositories (http and file) with order of priorities. I'll write a proposal explaining my full thoughts with more details later. It was a bit confuse to write it down in this pr.